### PR TITLE
Allow creation of new sites on account creation

### PR DESCRIPTION
**Pull request by ssundahlTTD**
A prior enhancement added the ability for admin users to create portal accounts and attach the participant to an existing site.  This enhancement allows creation of a new site for the participant.

image

This change also renames CheckboxInput to MultiCheckboxInput to make the purpose of that component more clear.

### CODE REVIEW

File: src/api/routers/participants/participantsCreation.ts [hunk 1]
```diff
@@ -50,19 +69,29 @@ export async function createParticipant(req: ParticipantRequest, res: Response)
   const types = await ParticipantType.query().findByIds(participantRequest.participantTypes);
   const apiRoles = await ApiRole.query().findByIds(participantRequest.apiRoles);
 
+  let site;
+  if (!participantRequest.siteId) {
+    // create site (UID2-2631)
```
Comment (alex-yau-ttd) [inline src/api/routers/participants/participantsCreation.ts:74]: We can remove this comment now? Or at least the reference to Jira

Comment (ssundahlTTD) [reply src/api/routers/participants/participantsCreation.ts:74]: removed

File: src/api/routers/participants/participantsCreation.ts [hunk 2]
```diff
@@ -50,19 +69,29 @@ export async function createParticipant(req: ParticipantRequest, res: Response)
...
+
   const participantData = ParticipantCreationAndApprovalPartial.parse({
     name: participantRequest.participantName,
     types,
     apiRoles,
     status: ParticipantStatus.Approved,
-    siteId: participantRequest.siteId,
+    siteId: participantRequest.siteId ?? site?.id,
     users: [user],
   });
```
[the remark this answers is not in the document]
Comment (ssundahlTTD) [reply src/api/routers/participants/participantsCreation.ts:91]: if addSite() fails, admin will return a non-200 response and the error handler will take care of it so it won't get to this code.  So it should be fine.

File: src/web/components/ParticipantManagement/AddParticipantDialog.tsx [hunk 3]
```diff
@@ -64,24 +68,42 @@ function AddParticipantDialog({
...
     }
   }, [open, reset]);
 
   const onSubmit = useCallback(
     async (formData: AddParticipantForm) => {
-      await onAddParticipant(formData);
+      const response = await onAddParticipant(formData);
+      if (response.status === 200) {
+        SuccessToast('Participant Added.');
+      }
```
[the remark this answers is not in the document]
Comment (ssundahlTTD) [reply src/web/components/ParticipantManagement/AddParticipantDialog.tsx:106]: The form actually shows the errors.  So the dialog will stay up and the error will show up at the top of the form.

File: src/api/routers/participants/participantsCreation.ts [hunk 4]
```diff
@@ -21,23 +35,44 @@ import {
   ParticipantCreationRequest,
 } from './participantClasses';
 
-export async function createParticipant(req: ParticipantRequest, res: Response) {
-  const participantRequest = ParticipantCreationRequest.parse(req.body);
+export async function validateParticipantCreationRequest(
```
Comment (alex-yau-ttd) [inline src/api/routers/participants/participantsCreation.ts:38]: nit: I wonder if it makes more sense to not export this function. Makes it clear that it isn't used outside this file. Thoughts?

**Review by alex-yau-ttd — APPROVED** [review]

**Pull request merged**